### PR TITLE
Refactor Type Names

### PR DESCRIPTION
**Pull request by batiati**
- Entirely ban `const Self = @This()` in favor of type names.
- Standardize the suffix `Type` for functions that build generic types, such as `fn ReplicaType(...) type` instead of `fn Replica(...) type`.

### CODE REVIEW

**Comment by matklad** [comment]
🔥 thank you so much for picking this up @batiati !

**Comment by batiati** [comment]
Btw, I didn't like the names `FIFOType` and `IOPSType`, but standards are to be followed!

**Comment by matklad** [comment]
>Btw, I didn't like the names FIFOType and IOPSType, but standards are to be followed!


Another option would be to change the standard iteself! Let't discuss this briefly in Slack, I am not entirely sure what I think about this!

File: src/lsm/set_associative_cache.zig [hunk 1]
```diff
@@ -85,7 +85,7 @@ pub fn SetAssociativeCacheType(
     assert(clock_hands_per_line > 0);
 
     return struct {
-        const Self = @This();
+        const BitIterator = @This();
```
Comment (matklad) [inline src/lsm/set_associative_cache.zig]: This is SetAssoativeCache, no?

File: src/lsm/set_associative_cache.zig [hunk 2]
```diff
@@ -699,15 +699,15 @@ fn PackedUnsignedIntegerArrayFuzzTest(comptime UInt: type) type {
     const testing = std.testing;
 
     return struct {
-        const Self = @This();
```
Comment (matklad) [inline src/lsm/set_associative_cache.zig:702]: same here

File: src/vsr/checkpoint_trailer.zig [hunk 3]
```diff
@@ -78,7 +78,7 @@ pub fn CheckpointTrailerType(comptime Storage: type) type {
         /// `encode_chunks()`/`decode_chunks()` return slices into this memory.
         block_bodies: [][]align(@sizeOf(u256)) u8,
 
-        // SoA representation of block references holding the trailer itself.
+        // SoA representation of block references holding the trailer itCheckpointTrailer.
```
Comment (matklad) [inline src/vsr/checkpoint_trailer.zig]: false positive!

File: src/vsr/clock.zig [hunk 4]
```diff
@@ -568,15 +568,15 @@ const DeterministicTime = @import("../testing/time.zig").Time;
 const DeterministicClock = ClockType(DeterministicTime);
 
 const ClockUnitTestContainer = struct {
-    const Self = @This();
+    const Clock = @This();
```
Comment (matklad) [inline src/vsr/clock.zig]: False positive

File: src/vsr/replica.zig [hunk 5]
```diff
@@ -107,29 +107,29 @@ pub fn ReplicaType(
     const GridScrubber = vsr.GridScrubberType(StateMachine.Forest);
 
     return struct {
-        const Self = @This();
+        const Replica = @This();
```
Comment (matklad) [inline src/vsr/replica.zig:110]: 💪

File: src/clients/c/test.zig [hunk 6]
```diff
@@ -61,13 +61,13 @@ fn RequestContextType(comptime request_size_max: comptime_int) type {
 
 // Notifies the main thread when all pending requests are completed.
 const Completion = struct {
-    const Self = @This();
+    const RequestContext = @This();
```
Comment (matklad) [inline src/clients/c/test.zig]: False positive

Comment (matklad) [reply src/clients/c/test.zig]: Could you also double-check the raw diff here? I fear I might have missed some!

File: src/lsm/set_associative_cache.zig [hunk 7]
```diff
@@ -614,18 +618,18 @@ fn PackedUnsignedIntegerArray(comptime UInt: type) type {
     assert(math.maxInt(BitsIndex) == uint_bits * (math.maxInt(WordIndex) + 1) - 1);
 
     return struct {
-        const Self = @This();
+        const BitIterator = @This();
```
Comment (matklad) [inline src/lsm/set_associative_cache.zig]: False positive

**Review by matklad — APPROVED** [review]

**Pull request merged**